framework/auth/core.py

```diff
@@ -1376,13 +1380,22 @@ def is_affiliated_with_institution(self, inst):
         return inst in self.affiliated_institutions
 
     def remove_institution(self, inst_id):
-        try:
-            self.affiliated_institutions.remove(inst_id)
-        except ValueError:
-            return False
-        return True
+        node_id = False
+        for node in self._affiliated_institutions:
```

Comment:
It would be better to iterate over the institutions in `self.affiliated_institutions` instead of the nodes in the protected `self._affiliated_institutions`.